src/lib/containers/discussion_forum/ForumThreadEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import { Button } from 'react-bootstrap'
+import {
+  useCreateThread,
+  useUpdateThreadTitle,
+  useUpdateThreadMessage,
+} from '../../utils/hooks/SynapseAPI/forum/useForum'
+import { CreateDiscussionThread } from '../../utils/synapseTypes/DiscussionBundle'
+import { MarkdownEditor } from '../markdown/MarkdownEditor'
+
+export type ForumThreadEditorProps = {
+  editTitle?: string
+  editText?: string
+  id: string
+  onCancel: () => void
```

Comment:
`onClose` is a better name since you also call it after saving

```suggestion
  onClose: () => void
```